src/HIE/Bios/Cradle.hs

```diff
@@ -493,37 +493,55 @@ withCabalWrapperTool (mbGhc, ghcArgs) wdir = do
 -- | Given the root directory, get the build dir we are using for cabal
 -- In the `hie-bios` cache directory
 cabalBuildDir :: FilePath -> IO FilePath
-cabalBuildDir work_dir = do
-  abs_work_dir <- makeAbsolute work_dir
+cabalBuildDir workDir = do
+  abs_work_dir <- makeAbsolute workDir
   let dirHash = show (fingerprintString abs_work_dir)
   getCacheDir ("dist-"<>filter (not . isSpace) (takeBaseName abs_work_dir)<>"-"<>dirHash)
 
+cabalGHCBin :: IO FilePath
+cabalGHCBin = do
+  result <- readProcessWithCwd "." "cabal" ["exec", "-v0", "--", "ghc", "--print-libdir"] ""
+  case result of
+    CradleFail err -> error (show err)
+    CradleNone     -> undefined
+    CradleSuccess r -> do
+      let strippedPath = T.unpack $ T.stripEnd $ T.pack r
+      let cabalGHC = strippedPath </> "../../bin/ghc"
```

Comment:
It seems the path in windows is different (only one `..`)?

```
PS D:\hls> cabal exec ghc -- --print-libdir
D:\bin\stack\x86_64-windows\ghc-8.8.4\lib
PS D:\hls> D:\bin\stack\x86_64-windows\ghc-8.8.4\lib\..\bin\ghc --version
The Glorious Glasgow Haskell Compilation System, version 8.8.4
```

Reply:
well that is certainly bad and answers whether the libdir is always relative to the ghc binary location